File: Python/8kyu/8kyu-double-char.py
# Double Char
# Language: Python
# Version: 3.10
# Difficulty: 8 kyu

# Solution 1
def double_char(s):
    split_s = [str(x) * 2 for x in str(s)]
    return "".join(split_s)

# Solution 2
def double_char(s):
    result = ""
    for i in s:
        result += i * 2
    return result

# Solution 3
def double_char(s):
    return "".join(i * 2 for i in s)


# Tests
test.assert_equals(double_char("String"),"SSttrriinngg")
test.assert_equals(double_char("Hello World"),"HHeelllloo  WWoorrlldd")
test.assert_equals(double_char("1234!_ "),"11223344!!__  ")

File: Julia/8kyu/8kyu-double-char.jl
# Double Char
# Language: Julia
# Version: 1.5
# Difficulty: 8 kyu

function doublechar(str)
  join(i^2 for i in str)
end


# Tests
using FactCheck

facts("Basic tests") do
  context("Should pass fixed tests") do
    @fact doublechar("String") --> "SSttrriinngg"
    @fact doublechar("Hello World") --> "HHeelllloo  WWoorrlldd"
    @fact doublechar("1234!_ ") --> "11223344!!__  "
  end
end